src/main/java/com/Model/UserDetail.java

```diff
@@ -0,0 +1,59 @@
+package com.Model;
+
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+public class UserDetail implements UserDetails {
+
+    private User user;
+    private List<GrantedAuthority> authorities;
+
+    public UserDetail(User user){
+        this.user = user;
+        this.authorities = new ArrayList<>();
+        if (user instanceof AdminUser){
+             this.authorities.add(new SimpleGrantedAuthority("ADMIN"));
+        } else {
+            this.authorities.add(new SimpleGrantedAuthority("USER"));
+        }
+    }
+
+    @Override
+    public Collection<? extends GrantedAuthority> getAuthorities() {
+        return this.authorities;
+    }
+
+    @Override
```

Comment:
I don't understand why these methods should be in two places. Do we really need that other User class?